src/nanotron/parallel/context.py

```diff
@@ -127,11 +127,35 @@ def set_device(self):
 
     def get_local_ranks(self, world_rank: int) -> Tuple[int, int, int]:
```

Comment:
Maybe add a type hint for which parallelism each rank corresponds to. For example [Tuple[Annotated[int, "pp rank"]], ...]